core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -293,6 +294,14 @@ public void onResponse(PrimaryShardReference primaryShardReference) {
                     final ShardRouting primary = primaryShardReference.routingEntry();
                     assert primary.relocating() : "indexShard is marked as relocated but routing isn't" + primary;
                     DiscoveryNode relocatingNode = clusterService.state().nodes().get(primary.relocatingNodeId());
+                    if (relocatingNode != null && relocatingNode.getVersion().major > Version.CURRENT.major) {
+                        // ES 6.x requires a primary context hand-off during primary relocation which is not implemented on ES 5.x.
```

Comment:
can add a sentence or two about what is currently known to be potentially missing? (in sync markers for shards on new nodes that should be accounted for GP calculations).  I think it will help (at least it would me) to understand what this is about.